Sources/OutlineView/TreeMap.swift

```diff
@@ -0,0 +1,223 @@
+
+/// An object representing the state of an OutlineView in which the content
+/// of the OutlineView is Identifiable. The TreeMap should be able to stay
+/// in sync with the OutlineView by reacting to data updates in the OutlineView.
+class TreeMap<D: Hashable> {
+        
+    struct Node {
+        var parentId: D?
+        var isLeaf: Bool
+        
+        // childIds can be nil if `isLeaf` == true, or if the item is not expanded.
+        // If childIds is not nil, that means the item is expanded.
+        var childIds: [D]?
```

Comment:
Does it make sense to coalesce `isLeaf` and `childIds` into a single enum?

```swift
enum State {
  case leaf
  case collapsed
  case expanded([D])
}
```

Reply:
Great idea. Changed that here: https://github.com/Sameesunkaria/OutlineView/pull/10/commits/9d2dc36710399aa415f24da069f110d9ed67345e